sssom/util.py

```diff
@@ -824,15 +824,13 @@ def get_prefixes_used_in_table(df: pd.DataFrame):
 
 def filter_out_prefixes(df: pd.DataFrame, filter_prefixes) -> pd.DataFrame:
     rows = []
+    
     for index, row in df.iterrows():
-        ok = True
-        for col in _defining_features:
-            v = row[col]
-            prefix = get_prefix_from_curie(v)
-            if prefix in filter_prefixes:
-                ok = False
-                break
-        if ok:
+        # Get list of CURIEs from the 3 columns (_defining_features) for the row.
+        list_of_curies = [get_prefix_from_curie(row_values) for row_values in row[_defining_features]]
+        # Confirm if none of the 3 CURIEs in the list above appear in the filter_prefixes list.
+        # If TRUE, append row.
+        if not any(curie in list_of_curies for curie in filter_prefixes):
```

Comment:
These variables are misleading as well. The loop variable should be `prefix` and not `curie`, since the variable `filter_prefixes` has a list of prefixes in it, not CURIEs..